src/app/core/services/label.service.ts

```diff
@@ -73,33 +75,72 @@ export class LabelService {
    * Stores the json data from Github api into the list of arrays in this service
    * @param labels: the json data of the label
    */
-  private populateLabelLists(labels: Array<{}>): void {
-
-    for (const label of labels) {
-      // Get the name and color of each label and store them into the service's array list
-      const labelName = String(label['name']).split('.');
-      const labelType = labelName[0];
-      const labelValue = labelName[1];
-      const labelColor = String(label['color']);
-
-      // Check for duplicate labels
-      if (this.labelColorMap.has(labelValue)) {
-        continue;
-      }
+  private storeLabelData(labels: Array<{}>): void {
 
-      this.labelColorMap.set(labelValue, labelColor);
-      const labelList = this.getLabelList(labelType);
+    // Parse Input Data to Label[]
+    const labelData: Label[] = this.parseLabelData(labels);
+    const expectedLabels: Label[] = Label.getRequiredLabels();
 
-      if (labelList !== undefined) {
-        labelList.push({labelValue: labelValue, labelColor: labelColor});
+    expectedLabels.forEach(label => {
```

Comment:
prefer to pass in all required data (e.g., `expectedLabels`) as parameters instead of accessing data directly. That way, the function becomes easier to understand/test.